rust/src/treefile.rs

```diff
@@ -1226,6 +1259,66 @@ etc-group-members:
                 == "table"
         );
     }
+
+    #[test]
+    fn test_split_whitespace_unless_quoted() {
+        // test single quoted package
+        let single_quoted_pkg = String::from("'foobar >= 1.0'");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&single_quoted_pkg).collect();
+        assert_eq!("foobar >= 1.0", pkgs[0]);
+
+        // test multiple quoted packages
+        let mult_quoted_pkg = String::from("'foobar >= 1.0' 'quuz < 0.5' 'corge > 2'");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&mult_quoted_pkg).collect();
+        assert_eq!("foobar >= 1.0", pkgs[0]);
+        assert_eq!("quuz < 0.5", pkgs[1]);
+        assert_eq!("corge > 2", pkgs[2]);
+
+        // test single unquoted package
+        let single_unquoted_pkg = String::from("foobar");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&single_unquoted_pkg).collect();
+        assert_eq!("foobar", pkgs[0]);
+
+        // test multiple unquoted packages
+        let mult_unquoted_pkg = String::from("foobar quuz corge");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&mult_unquoted_pkg).collect();
+        assert_eq!("foobar", pkgs[0]);
+        assert_eq!("quuz", pkgs[1]);
+        assert_eq!("corge", pkgs[2]);
+
+        // test different orderings of mixed quoted and unquoted packages
+        let mix_quoted_unquoted_pkgs = String::from("'foobar >= 1.1' baz-package 'corge < 0.5'");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&mix_quoted_unquoted_pkgs).collect();
+        assert_eq!("foobar >= 1.1", pkgs[0]);
+        assert_eq!("corge < 0.5", pkgs[1]);
+        assert_eq!("baz-package", pkgs[2]);
+        let mix_quoted_unquoted_pkgs = String::from("corge 'foobar >= 1.1' baz-package");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&mix_quoted_unquoted_pkgs).collect();
+        assert_eq!("foobar >= 1.1", pkgs[0]);
+        assert_eq!("corge", pkgs[1]);
+        assert_eq!("baz-package", pkgs[2]);
+        let mix_quoted_unquoted_pkgs =
+            String::from("corge 'foobar >= 1.1' baz-package 'quuz < 0.0.1'");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&mix_quoted_unquoted_pkgs).collect();
+        assert_eq!("foobar >= 1.1", pkgs[0]);
+        assert_eq!("quuz < 0.0.1", pkgs[1]);
+        assert_eq!("corge", pkgs[2]);
+        assert_eq!("baz-package", pkgs[3]);
+
+        // test missing quotes around packages using version qualifiers
+        let missing_quotes = String::from("foobar >= 1.0 quuz");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&missing_quotes).collect();
+        assert_ne!("foobar >= 1.0", pkgs[0]);
+        assert_eq!(">=", pkgs[1]);
```

Comment:
Then these tests would be:
`assert!(split_whitespace_unless_quoted(&missing_quotes).is_err())`